pkg/controller.v1/tensorflow/tfjob_controller.go

```diff
@@ -786,3 +833,28 @@ func (r *TFJobReconciler) createNewPod(tfjob *tfv1.TFJob, rt, index string, spec
 	}
 	return nil
 }
+
+// onOwnerCreateFunc modify creation condition.
+func onOwnerCreateFunc() func(event.CreateEvent) bool {
+	return func(e event.CreateEvent) bool {
+		tfJob, ok := e.Object.(*tensorflowv1.TFJob)
+		if !ok {
+			return true
+		}
+		scheme.Scheme.Default(tfJob)
+		msg := fmt.Sprintf("TFJob %s is created.", e.Object.GetName())
+		logrus.Info(msg)
+		//specific the run policy
+
+		if tfJob.Spec.RunPolicy.CleanPodPolicy == nil {
```

Comment:
Can we have one defaulting func for this?

Reply:
nice catch. Let me clean up this part.  We do have `SetDefaults_TFJob`